File: main/src/main/java/com/sedmelluq/discord/lavaplayer/player/hook/AudioOutputHook.java
package com.sedmelluq.discord.lavaplayer.player.hook;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.track.playback.AudioFrame;

/**
 * Hook for intercepting outgoing audio frames from AudioPlayer.
 */
public interface AudioOutputHook {
  /**
   * @param player Audio player where the frame is coming from
   * @param frame Audio frame
   * @return The frame to pass onto the actual caller
   */
  AudioFrame outgoingFrame(AudioPlayer player, AudioFrame frame);
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/player/hook/AudioOutputHook.kt
package com.sedmelluq.discord.lavaplayer.player.hook

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer
import com.sedmelluq.discord.lavaplayer.track.playback.AudioFrame

/**
 * Hook for intercepting outgoing audio frames from AudioPlayer.
 */
interface AudioOutputHook {
    /**
     * @param player Audio player where the frame is coming from
     * @param frame  Audio frame
     * @return The frame to pass onto the actual caller
     */
    fun outgoingFrame(player: AudioPlayer, frame: AudioFrame): AudioFrame
}
